app/routes/tasks.py

```diff
@@ -0,0 +1,94 @@
+from datetime import datetime
+from flask import Blueprint, jsonify, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from app.helpers import *
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# POST /tasks Create a new task to the db and return the task body.
+@tasks_bp.route("", methods = ["POST"])
+def post_new_task():
+
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return {"details": "Invalid data"}, 400
+    
+    new_task = Task(title=request_body["title"], description=request_body["description"], completed_at=request_body["completed_at"])
+    db.session.add(new_task)
+    db.session.commit()
+
+    return new_task.create_task_response(), 201
+
+# GET /tasks Get a list of all tasks. Handle sort requests via query params.
+@tasks_bp.route("", methods = ["GET"])
+def get_tasks():
+    sort_query = request.args.get("sort")
+    title_query = request.args.get("title")
+
+    if sort_query:
```

Comment:
Notice that if a user supplied both the `title` and the `sort` options, the title essentially overpowers the sort filter. Arguably, why would a user try to sort by title a list of all things with the same title, but consider if we loosen the title filtering to be a substring rather than an exact match. In that case, we might like to get the tasks containing some term, and then still sort the results.

Until we call a terminal query function, like `all()`, or `count()`, we can build up filters on the objects returned from the query instance, so we could `filter` and then `order_by`, depending on which params were available.